Libraries/LibJS/Runtime/TypedArrayPrototype.cpp

```diff
@@ -1433,7 +1433,11 @@ static ThrowCompletionOr<void> set_typed_array_from_typed_array(VM& vm, TypedArr
 
     // 16. If srcLength + targetOffset > targetLength, throw a RangeError exception.
     Checked<size_t> checked = source_length;
-    checked += static_cast<u32>(target_offset);
+
+    if (target_offset > static_cast<double>(NumericLimits<size_t>::max()))
```

Comment:
Eyeballing this and the surrounding code, it feels like we're not quite sure whether `u32` or `size_t` is the appropriate type for buffer storage size. This seems like a potential problem! Since `u32` would limit us to 4GB array buffers, maybe `size_t` is the right choice and we should go on to make sure it's used everywhere?
Not required to fix for this specific issue, but something to tackle afterwards perhaps :)